metrics/metrics.go

```diff
@@ -0,0 +1,79 @@
+package metrics
+
+import (
+	"github.com/go-kit/kit/metrics"
+	"github.com/go-kit/kit/metrics/multi"
+)
+
+// Registry has to implemented by any system that wants to monitor and expose metrics.
+type Registry interface {
+	// IsEnabled shows whether metrics instrumentation is enabled.
+	IsEnabled() bool
+	ReqsCounter() metrics.Counter
+	ReqDurationHistogram() metrics.Histogram
+	RetriesCounter() metrics.Counter
+}
+
+// NewMultiRegistry creates a new standardRegistry that wraps multiple Registries.
+func NewMultiRegistry(registries []Registry) Registry {
+	reqsCounters := []metrics.Counter{}
+	reqDurationHistograms := []metrics.Histogram{}
+	retriesCounters := []metrics.Counter{}
+
+	for _, r := range registries {
+		reqsCounters = append(reqsCounters, r.ReqsCounter())
+		reqDurationHistograms = append(reqDurationHistograms, r.ReqDurationHistogram())
+		retriesCounters = append(retriesCounters, r.RetriesCounter())
+	}
+
+	return &standardRegistry{
+		isEnabled:            true,
+		reqsCounter:          multi.NewCounter(reqsCounters...),
+		reqDurationHistogram: multi.NewHistogram(reqDurationHistograms...),
+		retriesCounter:       multi.NewCounter(retriesCounters...),
+	}
+}
+
+type standardRegistry struct {
+	isEnabled            bool
```

Comment:
could you rename `isEnabled` to `enable`?
But keep the method name `IsEnabled()`